From 9086784bf2d883b68dabf78a68164e3c71cff206 Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Tue, 5 May 2026 19:47:21 -0400
Subject: fix failure to report write error (ENOSPC) on fmemopen streams

---
 src/stdio/fmemopen.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/stdio/fmemopen.c b/src/stdio/fmemopen.c
index 343e3e3f..04472247 100644
--- a/src/stdio/fmemopen.c
+++ b/src/stdio/fmemopen.c
@@ -63,7 +63,12 @@ static size_t mwrite(FILE *f, const unsigned char *buf, size_t len)
 	}
 	if (c->mode == 'a') c->pos = c->len;
 	rem = c->size - c->pos;
-	if (len > rem) len = rem;
+	if (len > rem) {
+		errno = ENOSPC;
+		f->wpos = f->wbase = f->wend = 0;
+		f->flags |= F_ERR;
+		len = rem;
+	}
 	memcpy(c->buf+c->pos, buf, len);
 	c->pos += len;
 	if (c->pos > c->len) {
-- 
cgit v1.2.1

